### ISSUE TITLE
Silence the compiler for unused variables

### ISSUE THREAD

**Issue by sweihub** [stephenberry/glaze#177]
Howdy

Glaze for serializing/deserializing JSON is really awsome, would you please silence the compiler by removing the unused variables, thanks.

enable the compiler options `-Wunused-variable -Wunused-but-set-variable`

```
/home/swei/quant/trade/core/include/glaze/util/strod.hpp:404:22: warning: variable 'sig_end' set but not used [-Wunused-but-set-variable]
      const uint8_t *sig_end = nullptr;  /* significant part ending position */
...
/home/swei/quant/trade/core/include/glaze/util/dtoa.hpp:981:12: warning: unused variable 'start' [-Wunused-variable]
      auto start = buffer;
           ^
...

```

**Comment by mwalcott3** [comment]
@stephenberry and I were just talking about this. I thought he removed those today already.

Beyond this we should probably use -Wall -Wextra and -Werror in the unit tests so the ci builds will fail to make sure we are avoiding the common warnings.

**Comment by mwalcott3** [comment]
Addressed in #182

Glaze should no longer emit warnings with
-Wall -Wextra -pedantic
This is now enforced in the github workflow with -Werror for the unit tests so hopefully we shouldnt pollute other peoples builds with common warnings in the future.

MSVC is using /W4. There is still one warning left but that is an actual bug that needs to be fixed. Ill add /WX to the tests when that gets addressed.
